src/waveform/widgets/waveformwidgetinfo.h

```diff
@@ -0,0 +1,77 @@
+#pragma once
+
+#include "waveform/widgets/waveformwidgetcategory.h"
+#include "waveform/widgets/waveformwidgettype.h"
+
+class WaveformWidgetAbstract;
+
+class WaveformWidgetInfoBase {
+  public:
+    const WaveformWidgetType::Type m_type;
+    const QString m_name;
+    const bool m_useGL;
+    const bool m_useGLES;
+    const bool m_useGLSL;
+    const bool m_highDetail;
+    const WaveformWidgetCategory m_category;
```

Comment:
[CppCoreGuidelines C.12](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rc-constref): Don't make data members const or references in a copyable or movable type.
```suggestion
    WaveformWidgetType::Type m_type;
    QString m_name;
    bool m_useGL;
    bool m_useGLES;
    bool m_useGLSL;
    bool m_highDetail;
    WaveformWidgetCategory m_category;
```